comp/metadata/inventoryagent/status.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package inventoryagent
+
+import (
+	"embed"
+	htmlTemplate "html/template"
+	"io"
+	"path"
+	textTemplate "text/template"
+
+	"github.com/DataDog/datadog-agent/comp/core/status"
+)
+
+type statusProvider struct {
+	ia *inventoryagent
+}
+
+func newStatusProvider(ia *inventoryagent) statusProvider {
+	return statusProvider{
+		ia: ia,
+	}
+}
+
+func (s statusProvider) Name() string {
+	return "metadata"
+}
+
+func (s statusProvider) Index() int {
+	return 1
+}
+
+func (s statusProvider) getStatusInfo() map[string]interface{} {
+	return s.ia.Get()
```

Comment:
I would have expected the methods in this file to be part of the `inventoryagent` class directly. This way we could get rid of the dedicated `Get` method. WDYT ?

Reply:
Absolutely